lib/pacemaker/pcmk_sched_promotable.c

```diff
@@ -260,18 +260,22 @@ cmp_promotable_instance(gconstpointer a, gconstpointer b)
 
     CRM_ASSERT((rsc1 != NULL) && (rsc2 != NULL));
 
-    // Check sort index set by pcmk__set_instance_roles()
-    if (rsc1->sort_index > rsc2->sort_index) {
+    // Check promotion priority set by pcmk__set_instance_roles()
+    if (rsc1->private->promotion_priority > rsc2->private->promotion_priority) {
         pcmk__rsc_trace(rsc1,
-                        "%s has higher promotion priority than %s "
-                        "(sort index %d > %d)",
-                        rsc1->id, rsc2->id, rsc1->sort_index, rsc2->sort_index);
+                        "%s has higher promotion priority (%s) than %s (%d)",
+                        rsc1->id,
+                        pcmk_readable_score(rsc1->private->promotion_priority),
+                        rsc2->id, rsc2->private->promotion_priority);
```

Comment:
This is fine for now. Using `pcmk_readable_score()` for only one of them isn't ideal. One could be `-INFINITY` while the other one is `INFINITY`. Same for other cases.

Using readable scores for both without wasteful allocation in the common case would require making a string copy in a `pcmk__if_tracing()` block.

Reply:
yeah it's not worth it for debug or trace logs